__tests__/app.test.js

```diff
@@ -18,9 +19,85 @@ describe("/api/topics", () => {
       .then((response) => {
         const topicsArray = response.body.topics;
         expect(Array.isArray(topicsArray)).toBe(true);
+        // Ensuring topicsArray is not empty for handling false positive
+        expect(topicsArray.length).toBeGreaterThan(0);
         topicsArray.forEach((topic) => {
-          expect(topic).toHaveProperty("slug", expect.any(String));
-          expect(topic).toHaveProperty("description", expect.any(String));
+          // Use MatchObject instead toHaveProperty
+          expect(topic).toMatchObject({
+            slug: expect.any(String),
+            description: expect.any(String),
+          });
+        });
+      });
+  });
+});
+
+// Refactor of the Test as Requested in (PR#2)
+describe("/api", () => {
+  test("GET: 200 responds with an object describing all available endpoints", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((response) => {
+        const responseEndpoints = response.body.endpoints;
+        expect(responseEndpoints).toEqual(endpointsJSON);
+      });
+  });
+});
+
+describe("/api/articles/:article_id", () => {
+  test("GET: 200 responds with an article object including article_img_url", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then((response) => {
+        const article = response.body.article;
+        expect(article).toMatchObject({
+          author: "butter_bridge",
+          title: "Living in the shadow of a great man",
+          article_id: 1,
+          body: "I find this existence challenging",
+          topic: "mitch",
+          created_at: "2020-07-09T20:11:00.000Z",
+          votes: 100,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
+      });
+  });
+
+  test("GET: 404 responds with an error message for non-existent article", () => {
+    return request(app)
+      .get("/api/articles/999999")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("Article Not Found");
+      });
+  });
+});
+
+describe("/api/articles", () => {
+  test("GET: 200 responds with all an array of articles", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        const articles = response.body.articles;
+        expect(Array.isArray(articles)).toBe(true);
+        articles.forEach((article) => {
```

Comment:
You haven't added a length check here to make sure that your forEach is working as intended. If the array were empty this would still pass.